src/main/java/io/cdap/plugin/cdc/source/sqlserver/ResultSetToDMLRecord.java

```diff
@@ -58,14 +61,15 @@ public StructuredRecord call(ResultSet row) throws SQLException {
       .set(Schemas.OP_TYPE_FIELD, operationType.name())
       .set(Schemas.UPDATE_SCHEMA_FIELD, changeSchema.toString())
       .set(Schemas.UPDATE_VALUES_FIELD, getChangeData(row, changeSchema))
+      .set(Schemas.CHANGE_TRACKING_VERSION, row.getString("CHANGE_TRACKING_VERSION"))
       .build();
   }
 
   private static Map<String, Object> getChangeData(ResultSet resultSet, Schema changeSchema) throws SQLException {
     ResultSetMetaData metadata = resultSet.getMetaData();
     Map<String, Object> changes = new HashMap<>();
     for (int i = 0; i < changeSchema.getFields().size(); i++) {
-      int column = i + CHANGE_TABLE_COLUMNS_SIZE;
+      int column = 1 + i + CHANGE_TABLE_COLUMNS_SIZE;
```

Comment:
Not sure why we need 1+i+CHANGE_TABLE_COLUMNS_SIZE since we already update the col size to 4.

Reply:
it's because the things from ResultSetMetaData start from 1, while the schema fields start from 0. Will add a comment.

Reply:
if you're wondering why the 1 was added just in this PR, it's because the existing code is buggy -- it uses the wrong type for every single column... must have only been tested on strings or something.

Reply:
ah got it. Thanks for the comment there.